pkg/serverless/executioncontext/executioncontext.go

```diff
@@ -130,6 +133,13 @@ func (ec *ExecutionContext) UpdateOutOfMemoryRequestID(requestId string) {
 	ec.lastOOMRequestID = requestId
 }
 
+// UpdateRuntime updates the execution context with the runtime information
+func (ec *ExecutionContext) UpdateRuntime(runtime string) {
+	ec.m.Lock()
+	defer ec.m.Unlock()
+	ec.runtime = runtime
```

Comment:
I see the other functions in this file are locking and unlocking so this is probably right but i'm trying to grok why, my guess is there maybe multiple go routines that are using the execution context? Does this stop race conditions? 

could you help me understand why?

Reply:
Yes. 
There's a `sync.Mutex` [here](https://github.com/DataDog/datadog-agent/blob/joey/language-tag/pkg/serverless/executioncontext/executioncontext.go#L20C21-L20C31) for the executionContext.
If a struct contains a sync.Mutex, it means the struct is meant to be accessed safely by multiple goroutines concurrently. The sync.Mutex is used to protect critical sections of code to prevent data races and ensure exclusive access to the shared data.

Reply:
awesome! thanks